internal/aiptest/list/zero_page_size.go

```diff
@@ -0,0 +1,51 @@
+package list
+
+import (
+	"github.com/einride/protoc-gen-go-aip-test/internal/ident"
+	"github.com/einride/protoc-gen-go-aip-test/internal/onlyif"
+	"github.com/einride/protoc-gen-go-aip-test/internal/suite"
+	"github.com/einride/protoc-gen-go-aip-test/internal/util"
+	"github.com/stoewer/go-strcase"
+	"go.einride.tech/aip/reflect/aipreflect"
+	"google.golang.org/protobuf/compiler/protogen"
+)
+
+//nolint:gochecknoglobals
+var zeroPageSize = suite.Test{
+	Name: "zero page size",
+	Doc: []string{
+		"If page size is zero, server should use the default page size, and return that many results.",
+		"This test assume the default page size is <= 100.",
```

Comment:
Do we need to make this assumption on <= 100? Can we just not create a bunch of resources and do a a few list calls and expect that

* All calls return the same length of resources
* The length returned is above zero

Reply:
Not sure I understand.

Can we not do something like this (somewhat pseudo code)

```go
// assuming slice of several list calls called "resps"
if len(slices.Compact(resps)) != 1 || len(resps[0]) < 1 {
  // fail test
}
```

Reply:
I think 100 is reasonable, and I think skipping the test if that's not true is also reasonable 😄

Reply:
Ah. I missed that fact that 0 is actually allowed as a default value. Understood

Reply:
But maybe a simpler test then might just be to do various insert + list calls (where each list is called on a database of a different size) and expect their responses to be of the same size?

Reply:
I discussed with @thall that I think it makes sense to just not skip it anyway. The limit of 100 is reasonable, and if the default page size of the API under test is larger, then no pagination is being done. Assertion on the contents is still valuable.

There are other tests that explicitly test pagination, so we don't need to cover everything in this test.